src/java/picard/analysis/CollectWgsMetrics.java

```diff
@@ -334,9 +329,9 @@ public void addInfo(final SamLocusIterator.LocusInfo info, final ReferenceSequen
                 }
             }
 
-            final int depth = Math.min(readNames.size(), coverageCap);
-            if (depth < readNames.size()) basesExcludedByCapping += readNames.size() - coverageCap;
-            histogramArray[depth]++;
+            final int depth = Math.min(pileupSize, coverageCap);
+            if (depth < pileupSize) basesExcludedByCapping += pileupSize - coverageCap;
+            depthHistogramArray[depth]++;
```

Comment:
yes, there is no bug here (despite rumors to the contrary...) just an attempt at clarifying.